### PR TITLE
add schedule to hackathon page

### PR DESCRIPTION
**Pull request by mashehu**
expanded the javascript time conversion to be a less element agnostic, output different time formats and to highlight the current row.

### CODE REVIEW

File: public_html/events.php [hunk 1]
```diff
@@ -127,10 +127,14 @@ function sanitise_date_meta($event){
...
     <script type="text/javascript">
-    $("dd[data-timestamp]").each(function(){
+    $("[data-timestamp]").each(function(){
       var timestamp = $(this).data("timestamp");
+      var timeformat = $(this).data("timeformat") ? $(this).data("timeformat") : "HH:mm z, LL"; 
       var local_time = moment.tz(timestamp, "X", moment.tz.guess());
-      $(this).text(local_time.format("HH:mm z, LL"));
+      $(this).text(local_time.format(timeformat));
+      if(moment(timestamp,"X").diff(moment().format())<(30*60*1000)){ 
+        $(this).parent("tr").addClass("table-success"); // highlight row in schedule if current time is less than 30 minutes after time in row
```
Comment (ewels) [inline public_html/events.php:136]: 🚀

File: public_html/assets/css/nf-core-dark.css [hunk 2]
```diff
@@ -150,6 +150,19 @@ blockquote a {
 .table-bordered th{
   border-color: #454d55;
 }
+.table tr.table-success,
+.table .table-success>td{
```
Comment (ewels) [inline public_html/assets/css/nf-core-dark.css:154]: Is this needed? https://getbootstrap.com/docs/4.6/content/tables/#contextual-classes

File: markdown/events/2021/hackathon-march-2021.md [hunk 3]
```diff
@@ -83,4 +83,134 @@ More details to come, but so far our idea is to work in the following areas:
...
+                <th>Tuesday 23, 2021</th>
+                <th>Thursday 24, 2021</th>
+            </tr>
+        </thead>
+        <tbody>
+            <tr>
+                <td data-timestamp="1616403600" data-timeformat="HH:mm z">10:00</td>
+                <td>Welcome <br><a href="https://youtu.be/cuwcgZX6Li8"><i class="fab fa-youtube mr-1"></i>https://youtu.be/cuwcgZX6Li8</a><br> See Slack for Zoom link.</td>
+                <td>Check-in <br><a href="https://youtu.be/JdnG-QzZQdg"><i class="fab fa-youtube mr-1"></i>https://youtu.be/JdnG-QzZQdg</a><br> See Slack for Zoom link.</td>
+                <td>Check-in <br><a href="https://youtu.be/awmfwmPVnVI"><i class="fab fa-youtube mr-1"></i>https://youtu.be/awmfwmPVnVI</a><br> See Slack for Zoom link.</td>
```
Comment (ewels) [inline markdown/events/2021/hackathon-march-2021.md:101]:
```suggestion
                <td>Welcome <br><a href="https://youtu.be/cuwcgZX6Li8"><i class="fab fa-youtube mr-1"></i>https://youtu.be/cuwcgZX6Li8</a><br> See <a href="https://nfcore.slack.com/archives/C01LBUZCDEW">Slack <code>#hackathon-march-2021</code> channel</a> for Zoom link.</td>
                <td>Check-in <br><a href="https://youtu.be/JdnG-QzZQdg"><i class="fab fa-youtube mr-1"></i>https://youtu.be/JdnG-QzZQdg</a><br> See <a href="https://nfcore.slack.com/archives/C01LBUZCDEW">Slack <code>#hackathon-march-2021</code> channel</a> for Zoom link.</td>
                <td>Check-in <br><a href="https://youtu.be/awmfwmPVnVI"><i class="fab fa-youtube mr-1"></i>https://youtu.be/awmfwmPVnVI</a><br> See <a href="https://nfcore.slack.com/archives/C01LBUZCDEW">Slack <code>#hackathon-march-2021</code> channel</a> for Zoom link.</td>
```

File: markdown/events/2021/hackathon-march-2021.md [hunk 4]
```diff
@@ -83,4 +83,134 @@ More details to come, but so far our idea is to work in the following areas:
...
+                <td data-timestamp="1616416200" data-timeformat="HH:mm z">13:30</td>
+                <td>Hack!</td>
+                <td>Hack!</td>
+                <td>Hack!</td>
+            </tr>
+            <tr>
+                <td data-timestamp="1616418000" data-timeformat="HH:mm z">14:00</td>
+                <!-- <td></td>
+                <td></td>
+                <td></td> -->
```
Comment (ewels) [inline markdown/events/2021/hackathon-march-2021.md:164]: What's with the commented out table cells?

**Review by ewels — APPROVED** [review]
Quick read of the code LGTM 👍🏻

File: markdown/events/2021/hackathon-march-2021.md [hunk 5]
```diff
@@ -83,4 +83,134 @@ More details to come, but so far our idea is to work in the following areas:
...
+            <tr>
+                <td data-timestamp="1616412600" data-timeformat="HH:mm z">12:30</td>
+                <!-- <td></td>
+                <td></td>
+                <td></td> -->
+            </tr>
+            <tr>
+                <td data-timestamp="1616414400" data-timeformat="HH:mm z">13:00</td>
+                <td>Group catch-up <br>
+                  <a href="https://meet.jit.si/nf-core-hackathon-march-2021-dsl2-modules"><i class="fas fa-video mr-1"></i>DSL2-modules</a><br>
```
Comment (ewels) [inline markdown/events/2021/hackathon-march-2021.md:136]: Maybe `target="_blank"` on all of these?

File: markdown/events/2021/hackathon-march-2021.md (hunk 4 above)
Comment (mashehu) [reply markdown/events/2021/hackathon-march-2021.md:164]: I wasn't too sure, if I want them in or not. Decided now to not have them in 🙂 .

File: public_html/assets/css/nf-core-dark.css (hunk 2 above)
Comment (mashehu) [reply public_html/assets/css/nf-core-dark.css:154]: This is for the dark mode, where `.table-success` looks very outwashed.

**Comment by mashehu** [comment]
Now with automated ToCs for longer event pages 🎉 
Screenshot 2021-03-15 at 09 48 12
Had to work around a weird bug, where an inline html table somehow created too many closing divs (occured with the march hackathon page) 🤷‍♂️

File: public_html/assets/css/nf-core-dark.css (hunk 2 above)
Comment (ewels) [reply public_html/assets/css/nf-core-dark.css:154]: ahh 👍🏻

**Pull request merged**